crates/ruma-client-api/src/sync/sliding_sync_events.rs

```diff
@@ -0,0 +1,310 @@
+    pub sort: Option<Vec<String>>,
+
+    /// Sticky. Required state for each room returned. An array of event type and state key tuples.
+    /// Note that elements of this array are NOT sticky so they must be specified in full when they
+    /// are changed.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub required_state: Option<Vec<(RoomEventType, String)>>,
+
+    /// Sticky. The maximum number of timeline events to return per room
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub timeline_limit: Option<UInt>,
+
+    /// Sticky. Filters to apply to the list before sorting.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub filters: Option<Raw<SyncRequestListFilters>>,
+}
+
+/// The RoomSubscriptions of the SlidingSync Request
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+pub struct RoomSubscription {
+    /// Sticky. Required state for each room returned. An array of event type and state key tuples.
+    /// Note that elements of this array are NOT sticky so they must be specified in full when they
+    /// are changed.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub required_state: Option<Vec<(RoomEventType, String)>>,
+
+    /// Sticky. The maximum number of timeline events to return per room
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub timeline_limit: Option<UInt>,
+}
+
+/// Sliding Sync Request
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+pub struct SyncRequest {
+    /// The lists of rooms we're interested in
+    pub lists: Vec<Raw<SyncRequestList>>,
+
+    /// Specific rooms and event types that we want to receive events from
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub room_subscriptions: Option<BTreeMap<OwnedRoomId, RoomSubscription>>,
+
+    /// Specific rooms we no longer want to receive events from
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub unsubscribe_rooms: Option<Vec<OwnedRoomId>>,
+}
+
+ruma_api! {
+    metadata: {
+        description: "Get all new events in a sliding window of rooms since the last sync or a given point of time.",
+        method: POST,
+        name: "sync",
+        added: 1.0,
+        stable_path: "/_matrix/client/v3/sync",
+        // unstable_path: "/_matrix/client/unstable/org.matrix.msc3575/sync",
+        rate_limited: false,
+        authentication: AccessToken,
+    }
+
+    #[derive(Default)]
+    request: {
+        /// A point in time to continue a sync from.
+        ///
+        /// Should be a token from the `pos` field of a previous `/sync`
+        /// response.
+        #[serde(skip_serializing_if = "Option::is_none")]
+        #[ruma_api(query)]
+        pub pos: Option<&'a str>,
+
+        /// The maximum time to poll in milliseconds before returning this request.
```

Comment:
"in milliseconds" isn't relevant to users of Ruma, given the `Duration` type. Also "returning" seems a bit weird, maybe:

```suggestion
        /// The maximum time to poll before responding to this request.
```